[INFO   ] Logger: Record log in /storage/emulated/0/Python/Bakery/.kivy/logs/kivy_22-04-14_26.txt
[INFO   ] Kivy: v2.0.0
[INFO   ] Kivy: Installed at "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/__init__.py"
[INFO   ] Python: v3.9.7 (default, Oct  9 2021, 08:02:40) 
[GCC 11.1.0]
[INFO   ] Python: Interpreter at "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/bin/python3"
[INFO   ] Logger: Purge log fired. Analysing...
[INFO   ] Logger: Purge 54 log files
[INFO   ] Logger: Purge finished!
[INFO   ] Factory: 186 symbols loaded
[INFO   ] Image: Providers: img_tex, img_dds, img_sdl2 (img_pil, img_ffpyplayer ignored)
[INFO   ] Text: Provider: sdl2
[INFO   ] Window: Provider: sdl2
[INFO   ] GL: Using the "OpenGL ES 2" graphics system
[INFO   ] GL: Backend used <sdl2>
[INFO   ] GL: OpenGL version <b'OpenGL ES 3.1 v1.r7p0-02rel0.4ea33e6671ff828e33c99c982634cd65'>
[INFO   ] GL: OpenGL vendor <b'ARM'>
[INFO   ] GL: OpenGL renderer <b'Mali-T720'>
[INFO   ] GL: OpenGL parsed version: 3, 1
[INFO   ] GL: Texture max size <8192>
[INFO   ] GL: Texture max units <16>
[INFO   ] Window: auto add sdl2 input provider
[INFO   ] Window: virtual keyboard not allowed, single mode, not docked
[INFO   ] GL: NPOT texture support is available
[WARNING] Base: Unknown <android> provider
[INFO   ] Base: Start application main loop
[INFO   ] Base: Leaving application in progress...
[WARNING] stderr: Traceback (most recent call last):
[WARNING] stderr:   File "kivy/properties.pyx", line 861, in kivy.properties.ObservableDict.__getattr__
[WARNING] stderr: KeyError: 'products_tab'
[WARNING] stderr: 
[WARNING] stderr: During handling of the above exception, another exception occurred:
[WARNING] stderr: 
[WARNING] stderr: Traceback (most recent call last):
[WARNING] stderr:   File "/storage/emulated/0/Python/Bakery/__init__.py", line 29, in <module>
[WARNING] stderr:     Application().run()
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/app.py", line 950, in run
[WARNING] stderr:     runTouchApp()
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/base.py", line 582, in runTouchApp
[WARNING] stderr:     EventLoop.mainloop()
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/base.py", line 347, in mainloop
[WARNING] stderr:     self.idle()
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/base.py", line 387, in idle
[WARNING] stderr:     Clock.tick()
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/clock.py", line 733, in tick
[WARNING] stderr:     self.post_idle(ts, self.idle())
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/clock.py", line 776, in post_idle
[WARNING] stderr:     self._process_events()
[WARNING] stderr:   File "kivy/_clock.pyx", line 616, in kivy._clock.CyClockBase._process_events
[WARNING] stderr:   File "kivy/_clock.pyx", line 649, in kivy._clock.CyClockBase._process_events
[WARNING] stderr:   File "kivy/_clock.pyx", line 645, in kivy._clock.CyClockBase._process_events
[WARNING] stderr:   File "kivy/_clock.pyx", line 218, in kivy._clock.ClockEvent.tick
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/animation.py", line 353, in _update
[WARNING] stderr:     self.stop(widget)
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/animation.py", line 218, in stop
[WARNING] stderr:     self.dispatch('on_complete', widget)
[WARNING] stderr:   File "kivy/_event.pyx", line 705, in kivy._event.EventDispatcher.dispatch
[WARNING] stderr:   File "kivy/_event.pyx", line 1248, in kivy._event.EventObservers.dispatch
[WARNING] stderr:   File "kivy/_event.pyx", line 1172, in kivy._event.EventObservers._dispatch
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/arm-linux-androideabi/lib/python3.9/site-packages/kivy/uix/screenmanager.py", line 419, in _on_complete
[WARNING] stderr:     self.screen_in.dispatch('on_enter')
[WARNING] stderr:   File "kivy/_event.pyx", line 709, in kivy._event.EventDispatcher.dispatch
[WARNING] stderr:   File "/storage/emulated/0/Python/Bakery/Views/Home.py", line 23, in on_enter
[WARNING] stderr:     self.setup_tabs()
[WARNING] stderr:   File "/storage/emulated/0/Python/Bakery/Views/Home.py", line 28, in setup_tabs
[WARNING] stderr:     self.add_products()
[WARNING] stderr:   File "/storage/emulated/0/Python/Bakery/Views/Home.py", line 42, in add_products
[WARNING] stderr:     self.view.ids.products_tab.add_widget(product_frame)
[WARNING] stderr:   File "kivy/properties.pyx", line 864, in kivy.properties.ObservableDict.__getattr__
[WARNING] stderr: AttributeError: 'super' object has no attribute '__getattr__'
